yt_dlp/extractor/boomplay.py

```diff
@@ -0,0 +1,469 @@
+import base64
+import functools
+import json
+import re
+import urllib.parse
+
+from .common import InfoExtractor, SearchInfoExtractor
+from ..aes import aes_cbc_decrypt_bytes, aes_cbc_encrypt_bytes, unpad_pkcs7
+from ..utils import (
+    ExtractorError,
+    classproperty,
+    clean_html,
+    extract_attributes,
+    get_elements_text_and_html_by_attribute,
+    int_or_none,
+    join_nonempty,
+    merge_dicts,
+    orderedSet,
+    parse_count,
+    parse_duration,
+    strip_or_none,
+    unified_strdate,
+    url_or_none,
+    urlencode_postdata,
+    urljoin,
+    variadic,
+)
+from ..utils.traversal import traverse_obj
+
+
+class BoomplayBaseIE(InfoExtractor):
+    # Calculated from const values, see lhx.AESUtils.encrypt in public.js
+    # Note that the real key/iv differs from `lhx.AESUtils.key`/`lhx.AESUtils.iv`
+    _KEY = b'boomplayVr3xopAM'
+    _IV = b'boomplay8xIsKTn9'
+    _BASE = 'https://www.boomplay.com'
+    _MEDIA_TYPES = ('songs', 'video', 'episode', 'podcasts', 'playlists', 'artists', 'albums')
+    _GEO_COUNTRIES = ['NG']
+
+    @staticmethod
+    def __yield_elements_text_and_html_by_class_and_tag(class_, tag, html):
```

Comment:
since #10653 has been merged, should we use `..utils.traversal.find_element[s]` instead?

Reply:
yes. though also see #11402 about `find_element` (will be merged soon)

Reply:
https://github.com/yt-dlp/yt-dlp/blob/428ffb75aa3534b275cf54de42693a4d261519da/yt_dlp/utils/traversal.py#L404

didn't take a closer look into `find_element[s]`'s src. Seems like we still need this method

Reply:
Maybe #11402 should have changed that `assert` instead ... Matching `id` can/should ignore any `tag` (though there are wild pages with multiple instances of the same `id`) but the equivalent of s/t like CSS `div.player` ought to be supported.

There's also #5588 which could change this again/resolve this.

Upstream lacks the `get_element[s]_text_and_html_...()` variants, until the equivalent of #5588 is merged, which led me to create  `__yield_elements_text_and_html_by_class_and_tag()`.